nativelib/src/main/scala/java/lang/ieee754tostring/ryu/RyuDouble.scala

```diff
@@ -0,0 +1,548 @@
+// Copyright 2018 Ulf Adams
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Credits:
+//
+//    1) This work is a heavily modified derivation of the original work by
+//       Ulf Adams at URL: https://github.com/ulfjack/ryu.
+//       As such, it inherits the Apache license of the original work.
+//       Thank you Ulf Adams.
+//
+//    2) The original java sources were converted to a rough draft of
+//       scala code using the service at URL: javatoscala.com.
+//
+//       The raw conversion did not compile and contained bugs due
+//       to the handling of break and return statements, but it saved
+//       days, if not weeks, of effort.
+//
+//       Thank you javatoscala.com.
+//
+//    3) All additional work, including introduced bugs,  is an original
+//       contribution to Scala Native development.
+
+package java.lang.ieee754tostring.ryu
+
+import java.math.BigInteger
+
+import RyuRoundingMode._
+
+object RyuDouble {
+
+  var DEBUG: Boolean = false
```

Comment:
Do we really need this in our final code? I'd drop all debug printing altogether.

Reply:
I concur that final code should be as fast and light as correctness will allow, without
DEBUG or performance monitoring code.

Motivated by this review comment, I removed an obviously dead & worse than
useless dead DEBUG data structure (POW5) and its initialization at object
creation. Precious space & CPU cycles gained.

If you allow, I would like to keep the runtime printifs under the control
of the DEBUG variable around for a SN release or two, or at least
baselevels.  Agreed, they take up space in the generated code and
have a small runtime cost.  In addition, reviewers must skip over them.

I have two major reasons for wanting to keep them until this code is being
exercised by a few tens of users.  Basically, I am chicken:  Every new class
of user reveals a new class of bug.  I am reasonable confident that
that this code is at par with Ryu (which has a few known corner cases.)
but, based on fear not evidence, am concerned with least significant
digit issues.   I can verify reports a _lot_ quicker if I do not have to 
go back and figure out where the debugging writes should be.
Yeah, I could  write an awk script or such to strip the debugging
writes out of a private copy, but that is also error prone.  I agree
that leaving hanging "work to be done later" (a.k.a technical debt)
should be avoided in the general case.

My second reason for feeling more comfortable if the debugging
code is feft for a few evolutions is that I believe it may well help me
with backing out BigIntegers.

Please advise
with backing out BigIntegers or other alternatives.

Reply:
Debug is gone from current RyuDouble.scala. This facilitated the removal
of the dependency on BigInteger.  If/When RyuDouble.scala passes review,
I will make similar changes to RyuFloat.scala  which will allow this conversation
to be resolved.  For now it is still pending.

Reply:
Debug is gone from RyuFloat.scala